tsl/src/remote/async.c

```diff
@@ -473,16 +562,18 @@ get_single_response_nonblocking(AsyncRequestSet *set)
 		PGconn *pg_conn = remote_connection_get_pg_conn(req->conn);
 
 		if (remote_connection_is_processing(req->conn) && req->state == DEFERRED)
-			elog(ERROR,
-				 "can't get a response because previous request hasn't completed. Aborted SQL "
-				 "statement: %s",
-				 req->sql);
+			return async_response_error_create("request already in process");
```

Comment:
I didn't feel like the previous message followed PG error style ("keeping the primary message short helps keep it to the point") and this really shouldn't happen so seems like a corner-case. I think the most important thing is ability to find the place in the code where it happens, if it would ever occur. I think it is unique enough for that.